### ISSUE TITLE
Fix inspect.signature() of operator.{attrgetter,itemgetter,methodcaller} instances

### ISSUE THREAD

**Issue by dgilland** [python/cpython#118285]
# Bug report

### Bug description:

On Python 3.12.3 and 3.11.9, it looks like the `inspect.Signature` object for operator's `attrgetter`, `itemgetter`, and `methodcaller` classes doesn't match their `__call__` method signature:

```pycon
>>> import inspect
>>> import operator
>>> inspect.signature(operator.attrgetter("spam"))
<Signature (*args, **kwargs)>
>>> inspect.signature(operator.itemgetter("spam"))
<Signature (*args, **kwargs)>
>>> inspect.signature(operator.methodcaller("spam"))
<Signature (*args, **kwargs)>
```

but their `__call__` methods only accept a single argument:

```python
# attrgetter / itemgetter
    def __call__(self, obj):
        return self._call(obj)

# methodcaller
    def __call__(self, obj):
        return getattr(obj, self._name)(*self._args, **self._kwargs)
```

### CPython versions tested on:

3.11, 3.12

### Operating systems tested on:

macOS

### Linked PRs
* gh-118316

**Comment by terryjreedy** [comment]
The signatures of the getter/caller classes are the signature of their `.__init__` methods, and inspect.signature returns these properly.  The signature of callable instances should, it seems, match that of the `.__call__` methods, but, as you tested, it does not.  However, while the operator callables (functions and these 3 classes) are all initially defined in python, they are all replaced with C implementations, when possible, with `try: from _operator import *`.  After this, `itemgetter.__call__`, for instance, is a 'wrapper_descriptor' and `itemgetter(s).__call__` is a 'method_wrapper'.  Perhaps the C code for the call methods was never revised to use argument-clinic.

**Comment by serhiy-storchaka** [comment]
Actually, the `operator` module was initially implemented in C. The Python implementation was added later, partially as a fallback for alternative Python implementations (although Jython, PyPy, and other already had their own implementations), partially as experiment (which failed in my opinion).

Now `inspect.signature()` returns `<Signature (*args, **kwargs)>` for custom callable objects implemented in C instead of raising a ValueError. It was perhaps not intentional change, but it may be too later to change this again, some code can already rely on this. At least not in 3.11 (which is in the security fixes mode) or 3.12.

In any case, I added a simple way to specify signature for callables implemented in C and implemented it for the `operator` callables. Whether make `inspect.signature()` raising a ValueError by default is an open question.
